docs/source/working/processes.rst

```diff
@@ -123,9 +123,19 @@ An example input port that explicitly sets all these attributes is the following
 
 .. code:: python
 
-    spec.input('positive_number', required=False, default=Int(1), valid_type=(Int, Float), validator=is_number_positive)
+    spec.input('positive_number', required=False, default=lambda: Int(1), valid_type=(Int, Float), validator=is_number_positive)
+
+Here we define an input named ``positive_number`` that is not required, if a value is not explicitly passed, the default will be used and if a value *is* passed, it should be of type ``Int`` or ``Float`` and it should be valid according to the ``is_number_positive`` validator.
+
+.. warning::
+
+    In python, it is important to `never use mutable defaults for function arguments <https://docs.python.org/3/reference/compound_stmts.html#function-definitions>`_, since they are instantiated at function definition and reused for each invocation.
```

Comment:
```suggestion
    In python, it is good practise to avoid mutable defaults for function arguments, `since they are instantiated at function definition and reused for each invocation. <https://docs.python.org/3/reference/compound_stmts.html#function-definitions>`_
```